sdk/core/core-amqp/package.json

```diff
@@ -76,7 +76,7 @@
   },
   "devDependencies": {
     "@azure/eslint-plugin-azure-sdk": "^2.0.1",
-    "@azure/identity": "1.1.0-preview1",
```

Comment:
core-amqp will be releasing before identity goes out. Can we skip updating this dep here?
cc @chradek

Reply:
Same for the other core packages

Reply:
Approving this PR as this change should have gone ahead a few weeks ago.
@chradek, Please take a call on whether the core packages should stick to GA or last preview of Identity in a separate PR

Reply:
since it's a dev dep, it shouldn't matter as the only time it'll get installed is cloning from source

Reply:
I think it's fine for core packages to use the preview version of identity as a devDependency because it doesn't affect runtime logic, and we can test samples using the GA version of identity.

Reply:
should i still update something here?

Reply:
Wouldn't `npm install` on `core-amqp` fail if the dev dep points to identity preview 2 which hasn't been released yet?

Reply:
@KarishmaGhiya, please go ahead with merging this PR. We can continue the above discussion out of scope of this PR

Reply:
> Wouldn't `npm install` on `core-amqp` fail if the dev dep points to identity preview 2 which hasn't been released yet?

If you did it from inside the source tree and didn't do `rush update` first, then yes. But if you installed the package with `npm install @azure/core-amqp` then no, since it wouldn't install the devdeps

Reply:
Getting test failures due to the new identity preview - does a new dependency need to be added? @jonathandturner 

```
(!) Unresolved dependencies
https://rollupjs.org/guide/en/#warning-treating-module-as-external-dependency
constants (imported by dist-esm/test/crypto.test.js, dist-esm/src/cryptographyClient.js)
child_process (imported by ../../identity/identity/dist-esm/src/credentials/azureCliCredential.js)
stream (imported by ../../../common/temp/node_modules/.pnpm/registry.npmjs.org/jws/3.2.2/node_modules/jws/lib/sign-stream.js, ../../../common/temp/node_modules/.pnpm/registry.npmjs.org/jws/3.2.2/node_modules/jws/lib/verify-stream.js, stream?commonjs-external)
util (imported by ../../../common/temp/node_modules/.pnpm/registry.npmjs.org/jws/3.2.2/node_modules/jws/lib/sign-stream.js, ../../../common/temp/node_modules/.pnpm/registry.npmjs.org/jws/3.2.2/node_modules/jws/lib/verify-stream.js, util?commonjs-external)
[!] Error: Unexpected character '' (Note that you need plugins to import files that are not JavaScript)
../../../common/temp/node_modules/.pnpm/registry.npmjs.org/keytar/5.4.0/node_modules/keytar/build/Release/keytar.node (1:0)
1: ELF>�;@x�@8@}} �� � (
   ^
2: 0
3:  8�8� 8� PPP�td�t�t�t��Q�tdR�td�� � ��%P	i�!�(�#Q`��J�LN%(�I��%�r)@� ��e@K $.�PSTWY\abfghmpqrstwy{|��������k6��9Nzw������Mr#�z:2ôM�!�0���
Error: Unexpected character '' (Note that you need plugins to import files that are not JavaScript)
    at error (/home/vsts/work/1/s/common/temp/node_modules/.pnpm/registry.npmjs.org/rollup/1.32.1/node_modules/rollup/dist/shared/node-entry.js:5400:30)
    at Module.error (/home/vsts/work/1/s/common/temp/node_modules/.pnpm/registry.npmjs.org/rollup/1.32.1/node_modules/rollup/dist/shared/node-entry.js:9824:16)
    at tryParse (/home/vsts/work/1/s/common/temp/node_modules/.pnpm/registry.npmjs.org/rollup/1.32.1/node_modules/rollup/dist/shared/node-entry.js:9717:23)
    at Module.setSource (/home/vsts/work/1/s/common/temp/node_modules/.pnpm/registry.npmjs.org/rollup/1.32.1/node_modules/rollup/dist/shared/node-entry.js:10080:33)
    at Promise.resolve.catch.then.then.then (/home/vsts/work/1/s/common/temp/node_modules/.pnpm/registry.npmjs.org/rollup/1.32.1/node_modules/rollup/dist/shared/node-entry.js:12366:20)
    at <anonymous>
```